### PR TITLE
Variables.md added in kotlin folder 

### LINKED ISSUE #1385: [Concept Entry] Kotlin: Variables
### Reason/inspiration (optional)

We would like a new concept entry for [Kotlin](https://www.codecademy.com/resources/docs/kotlin) about variables.

The entry should go in a new file in docs/content/kotlin/concepts/variables/variables.md

The entry should

- Explain what variables are.
- Explain the syntax of declaring a variable.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by dakshdeepHERE**
### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

Fixes #1385

### CODE REVIEW

**Comment by dakshdeepHERE** [comment]
I think I created this variable branch under data-types branch but no worries I have deleted the data types files before pushing this variable file so the merge won't conflict.

File: content/kotlin/concepts/variables/variables.md
Comment (SSwiniarski) [inline content/kotlin/concepts/variables/variables.md:68]: Can we have an example of the `prop` property in use?

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@dakshdeepHERE I did a first review and left a few comments.

File: content/kotlin/concepts/variables/variables.md
Comment (dakshdeepHERE) [reply content/kotlin/concepts/variables/variables.md:68]: yes sure you want me to create a new header or just a code sample?

**Comment by dakshdeepHERE** [comment]
@SSwiniarski I completed all changes just have one question that I mentioned in your comment please check it out.

File: content/kotlin/concepts/variables/variables.md
Comment (SSwiniarski) [reply content/kotlin/concepts/variables/variables.md:68]: @dakshdeepHERE, you can just extend the existing example.

Comment (dakshdeepHERE) [reply content/kotlin/concepts/variables/variables.md:68]: alright @SSwiniarski

**Comment by dakshdeepHERE** [comment]
@SSwiniarski added more info for `prop`.

**Review by SSwiniarski — APPROVED** [review]
@dakshdeepHERE looks good for a second review.

**Comment by dakshdeepHERE** [comment]
thank you @SSwiniarski

File: content/kotlin/concepts/variables/variables.md
Comment (Dusch4593) [inline content/kotlin/concepts/variables/variables.md:19]:
```suggestion

```

File: content/kotlin/concepts/variables/variables.md
Comment (Dusch4593) [inline content/kotlin/concepts/variables/variables.md:38]:
```suggestion
In Kotlin, the `var` keyword is used to declare mutable variable while the `val` keyword is used for a read-only variable. The variable can either be explicitly typed, as shown with `x`. The type can also be inferred, as shown with `y`.
```

File: content/kotlin/concepts/variables/variables.md
Comment (Dusch4593) [inline content/kotlin/concepts/variables/variables.md:40]:
```suggestion
> **Note:** Unlike other programming languages, Kotlin doesn't have a `null` type. Instead, it has nullable and non-nullable types. A nullable type can hold a `null` value, while a non-nullable type cannot. To declare a nullable type, the `?` operator is used after the type. For example:
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@dakshdeepHERE I left some comments/suggestions for you. 😄 

Starting on line # 49, this remaining content would go better in the entry about classes (#1421), including the text and \```kotlin example snippets. Could we do the following?

- Remove this content from this PR and transfer to somewhere in the file in #1421.
- Replace this content with an example that just demonstrates declaring variables that are mutable, read-only, nullable, etc.

**Comment by dakshdeepHERE** [comment]
done @Dusch4593

**Review by Dusch4593 — APPROVED** [review]
@dakshdeepHERE LGTM for merge. 👍🏻

I made some last-minute edits in a0ac5839ce2136ee19d0293f9ad90e0b233b92db before doing so.

**Comment by Dusch4593** [comment]
Thank you for another wonderful contribution to Codecdemy Docs, @dakshdeepHERE! 🚀 🎊 🎉 

The new entry should now be live: https://www.codecademy.com/resources/docs/kotlin/variables

Screen Shot 2023-01-18 at 6 15 55 PM

Looking forward to your next Docs contribution! 😄

**Pull request merged**
